Modules/FindCUDA.cmake

```diff
@@ -683,6 +684,24 @@ find_path(CUDA_TOOLKIT_INCLUDE
 find_path(CUDA_TOOLKIT_INCLUDE device_functions.h)
 mark_as_advanced(CUDA_TOOLKIT_INCLUDE)
 
+if (CUDA_VERSION VERSION_GREATER "7.0")
+  set(cuda_fp16_found ${CUDA_TOOLKIT_INCLUDE})
+else()
+  find_path( cuda_fp16_found
+    cuda_fp16.h # Header included in toolkit
+    PATHS ${CUDA_TOOLKIT_INCLUDE}
+    ENV CUDA_PATH
+    ENV CUDA_INC_PATH
+    NO_DEFAULT_PATH
+    )
```

Comment:
Sure let's drop the cache. However, EXISTS was my first try (here: https://github.com/borisfom/CMake/commit/3da716ac0039f1f73cba5ab54d3adf9804d17058) but it would always return false. Was it because of odd operator precedence ?